### DISCUSSION TITLE
Faster indexing alternatives

### DISCUSSION THREAD

**Discussion by vyasr** [junegunn/fzf#2403] [Q&A]
`fzf` is a a fantastic tool, and it's generally fast enough for my purposes. However, in some cases where I'm indexing projects with 10s to 100s of thousands of files, Ctrl+T (or the `fzf.vim` equivalent, `:Files`) is bottlenecked by the speed of the `find` command used to generate the list of files. As already discussed on #763, this isn't `fzf`'s fault at all, and can be fixed by using an alternative `FZF_CTRL_T_COMMAND`.

I'm curious what practical recommendations for such a command would be. In particular, when I'm working on machines other than my own (i.e. where I don't have root permissions), I have little control over the `mlocate` database, so `locate` isn't an option. What other alternatives do/have people use(d)? One option would be to simply cache a local "database" by running the find command and storing the output to a file, then having `fzf` just parse that file instead of searching. Are there better options? Another good alternative is `git ls-files` (which is bound to `:GFiles` in `fzf.vim`) when working in a git repo.

Thanks in advance!

**Comment by timacdonald** [comment]
I use: https://github.com/sharkdp/fd

**Reply by vyasr** [reply]
Thanks! That's a good find, I hadn't seen `fd` before.
